sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -73,18 +73,19 @@ public TimeSeriesInsightsClient(string environmentFqdn, TokenCredential credenti
         /// </remarks>
         public TimeSeriesInsightsClient(string environmentFqdn, TokenCredential credential, TimeSeriesInsightsClientOptions options)
         {
+            Argument.AssertNotNullOrEmpty(environmentFqdn, nameof(environmentFqdn));
+            Argument.AssertNotNull(credential, nameof(environmentFqdn));
```

Comment:
nameof(credential). Why not the same check - AssertNotNullOrEmpty?

Reply:
Good catch on the 'nameof'.

The reason why it's AssertNotNull and not AssertNotNullOrEmpty is because the latter is used for strings, and credential is a TokenCredential type.